zhaquirks/inovelli/VZM31SN.py

```diff
@@ -1,578 +1,14 @@
-            },
-        },
-    }
-
-    device_automation_triggers = INOVELLI_AUTOMATION_TRIGGERS
-
-
-class InovelliVZM31SN(CustomDevice):
-    """VZM31-SN 2 in 1 Switch/Dimmer Module."""
-
-    signature = {
-        MODELS_INFO: [("Inovelli", "VZM31-SN")],
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Identify.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    Metering.cluster_id,
-                    ElectricalMeasurement.cluster_id,
-                    INOVELLI_VZM31SN_CLUSTER_ID,
-                ],
-                OUTPUT_CLUSTERS: [Identify.cluster_id, Ota.cluster_id],
-            },
-            2: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
-                INPUT_CLUSTERS: [Identify.cluster_id],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    INOVELLI_VZM31SN_CLUSTER_ID,
-                ],
-            },
-        },
-    }
-
-    replacement = {
-        ENDPOINTS: {
-            1: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMABLE_LIGHT,
-                INPUT_CLUSTERS: [
-                    Basic.cluster_id,
-                    Identify.cluster_id,
-                    Groups.cluster_id,
-                    Scenes.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    Metering.cluster_id,
-                    ElectricalMeasurement.cluster_id,
-                    Diagnostic.cluster_id,
-                    InovelliVZM31SNCluster,
-                ],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    Ota.cluster_id,
-                    InovelliVZM31SNCluster,
-                ],
-            },
-            2: {
-                PROFILE_ID: zha.PROFILE_ID,
-                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
-                INPUT_CLUSTERS: [Identify.cluster_id],
-                OUTPUT_CLUSTERS: [
-                    Identify.cluster_id,
-                    OnOff.cluster_id,
-                    LevelControl.cluster_id,
-                    InovelliVZM31SNCluster,
-                ],
-            },
-        },
-    }
-
-    device_automation_triggers = INOVELLI_AUTOMATION_TRIGGERS
+(
+    QuirkBuilder("Inovelli", "VZM31-SN")
+    .replaces(InovelliVZM31SNCluster)
+    .replaces(InovelliVZM31SNCluster, endpoint_id=2, cluster_type=ClusterType.Client)
```

Comment:
I think we should be able to use `replace_cluster_occurrences` from https://github.com/zigpy/zigpy/pull/1441 for the custom output clusters on all endpoints.

Haven't tested this, but this might work (if zigpy is new enough on the quirks repo, might need to be bumped?):
```suggestion
    .replace_cluster_occurrences(InovelliVZM31SNCluster, cluster_type=ClusterType.Client)
```